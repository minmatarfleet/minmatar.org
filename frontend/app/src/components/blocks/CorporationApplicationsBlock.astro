---
import { getLangFromUrl, useTranslations } from '@i18n/utils';

const lang = getLangFromUrl(Astro.url);
const t = useTranslations(lang);

import type { CorporationApplications } from '@dtypes/layout_components'

interface Props {
    corporation:  CorporationApplications;
    [propName: string]: any;
}

const {
    corporation,
    ...attributes
} = Astro.props

import BlockList from '@components/compositions/BlockList.astro';

import CorporationBadge from '@components/blocks/CorporationBadge.astro';
import CorporationApplicationItem from '@components/blocks/CorporationApplicationItem.astro';
import ComponentBlockHeader from '@components/blocks/ComponentBlockHeader.astro';
---

<div
    {...attributes}
    x-data={`{
        show_corporation() {
            return ("${corporation.corporation_id}" in corporations_filtered) && corporations_filtered[${corporation.corporation_id}].length > 0
        },
        get_applications_count() {
            if ("${corporation.corporation_id}" in corporations_filtered_applications_count)
                return corporations_filtered_applications_count[${corporation.corporation_id}]

            return 0
        }
    }`}
    x-show="show_corporation()"
>
    <ComponentBlockHeader
        id={corporation.corporation_id.toString()}
    >
        <CorporationBadge
            slot="header"
            corporation={{
                id: corporation.corporation_id,
                name: corporation.corporation_name,
                size: 'lg',
            }}
        >
            <small
                x-text={'`${get_applications_count()} ${get_applications_count() != 1 ? `'+t('applications').toLowerCase()+'` : `'+t('application').toLowerCase()+'`}`'}
            >
                {corporation.applications.length} {corporation.applications.length != 1 ? t('applications').toLowerCase() : t('application').toLowerCase()}
            </small>
        </CorporationBadge>

        <BlockList
            gap='var(--space-3xs)'
            class="[ min-h-[64px] justify-center ]"
            x-show="expanded"
            x-collapse.duration.100ms
        >
            {corporation.applications.map( (application) => 
                <CorporationApplicationItem
                    application={application}
                    corporation={{
                        id: corporation.corporation_id,
                        name: corporation.corporation_name,
                    }}
                    x-data={`{
                        id: $id('corporation-application-item'),
                        show_item() {
                            return ("${corporation.corporation_id}" in corporations_filtered) && corporations_filtered[${corporation.corporation_id}].includes(${application.id})
                        }
                    }`}
                    x-show="show_item()"
                />
            )}
        </BlockList>
    </ComponentBlockHeader>
</div>