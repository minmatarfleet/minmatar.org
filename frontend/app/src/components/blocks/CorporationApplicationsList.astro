---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import type { Alert, CorporationApplications } from '@dtypes/layout_components'

interface Props {
    corporations:   CorporationApplications[];
    alert?:         Alert | false;
    [propName: string]: any;
}

const {
    corporations,
    alert = false,
    ...attributes
} = Astro.props

import CorporationApplicationsBlock from '@components/blocks/CorporationApplicationsBlock.astro';
import TextBox from '@components/layout/TextBox.astro';
import Flexblock from '@components/compositions/Flexblock.astro';
---

<div
    id="corporations-applications-list"
    class="[ corporations-applications-list ]"
    {...attributes}
>
    {corporations.length > 0 &&
        <Flexblock class="[ correct-gaps ]">
            <p x-show="total_applications_filtered == 0" style="display: none;">{t('no_applications')}</p>

            {corporations.map( (corporation) => 
                <CorporationApplicationsBlock corporation={corporation} />
            )}
        </Flexblock>
    }

    {corporations.length == 0 &&
        <slot>
            <TextBox>
                <p>{t('no_applications')}</p>
            </TextBox>
        </slot>
    }
</div>