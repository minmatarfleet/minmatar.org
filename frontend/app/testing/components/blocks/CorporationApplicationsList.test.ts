import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test, vi } from "vitest";
import CorporationApplicationsList from "@components/blocks/CorporationApplicationsList.astro";

import CorporationApplicationsBlock from "@components/blocks/CorporationApplicationsBlock.astro";
vi.mock("@components/blocks/CorporationApplicationsBlock.astro");

test("CorporationApplicationsList defaults", async () => {
  vi.mocked(CorporationApplicationsBlock).mockResolvedValue(
    "CorporationApplicationsBlock_component"
  );
  const container = await AstroContainer.create();
  const result = await container.renderToString(CorporationApplicationsList, {
    props: {
      corporations: [
        {
          corporation_id: 12,
          applications: [],
        },
        {
          corporation_id: 11432,
          applications: [],
        },
      ],
    },
  });

  // expect(result).toMatchSnapshot();
});
