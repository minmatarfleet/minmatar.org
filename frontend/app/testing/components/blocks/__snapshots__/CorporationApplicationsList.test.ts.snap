// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CorporationApplicationsList defaults 1`] = `"<div id="corporations-applications-list" class="[ corporations-applications-list ]"> <div class="flex-block [ correct-gaps ] gap---space-s" data-mod style="--flex_block_gap: var(--space-s);">  <p x-show="total_applications_filtered == 0" style="display: none;">No applications found.</p> CorporationApplicationsBlock_componentCorporationApplicationsBlock_component </div>   </div>"`;
